Update components, add x-checker-data

PunkPangolin opened the pull request.

PunkPangolin committed:
Update components, add x-checker-data

PunkPangolin committed:
Download from Gitlab

hnm-duncan commented:
Thanks! Very late to the show, but I have uploaded the broken screenshots to gitlab registry. Another PR is pending to fix the links so that your PR passes and can be merged!

Tuxfamily has been great, but it is sadly going away now, it appears the web host is also down at the moment. I will have to look at an alternative hosting place.

PunkPangolin committed:
Update runtime to 25.08

PunkPangolin committed:
Fix x-checker-data

PunkPangolin commented:
> Thanks! Very late to the show, but I have uploaded the broken screenshots to gitlab registry. Another PR is pending to fix the links so that your PR passes and can be merged!

Great! I also updated the runtime to the recently released 25.08 and fixed the x-checker-data. Now everything should work (hopefully).

hnm-duncan commented:
Can we use the git checker for the x-checker-data of hikounomizu? It should be more reliable than the website (especially now that it is down ahah), and there will always be a v[...] tag with a semver version on gitlab for every release.

hnm-duncan commented:
I think the 25.08 freedesktop sdk will fail now because the SDL2_image dependency was removed. It needs to be bundled back manually, I can do it in a separate MR if you don't want to deal with that on this PR.

I also plan to upgrade to SDL3 soon so we could just leave the 24.08 in the meantime as an alternative.

PunkPangolin committed:
Revert runtime to 24.08

PunkPangolin commented:
> Can we use the git checker for the x-checker-data of hikounomizu? It should be more reliable than the website (especially now that it is down ahah), and there will always be a v[...] tag with a semver version on gitlab for every release.

I played around with it and failed. I think you can use the git checker only to download tags, not for a version query for a URL. The most elegant way would probably be to use JSON and the REST API, but I couldn't get that to work either.

> I think the 25.08 freedesktop sdk will fail now because the SDL2_image dependency was removed. It needs to be bundled back manually, I can do it in a separate MR if you don't want to deal with that on this PR.
> I also plan to upgrade to SDL3 soon so we could just leave the 24.08 in the meantime as an alternative.

SDL2 is available in the shared-modules, but as long as 24.08 is supported, there is no need for that, and staying with 24.08 is likely the easiest.

hnm-duncan commented:
Cool, thanks!

hnm-duncan commented:
Just curious about the new build for enet, using direct autoreconf instead of buildsystem: autotools. Is it because the buildsystem approach did not work with the github enet release link? I saw it is a bit lighter that the bespin version (does not include the docs).

PunkPangolin commented:
> Just curious about the new build for enet, using direct autoreconf instead of buildsystem: autotools. Is it because the buildsystem approach did not work with the github enet release link? I saw it is a bit lighter that the bespin version (does not include the docs).

The build still uses autotools; `buildsystem: autotools` is default if nothing else is declared. The autoreconf command is needed when downloading the sources from GitHub but not for the website download. To be honest, I cannot remember why I changed that to GitHub – if you prefer, I can revert it.

hnm-duncan commented:
> > Just curious about the new build for enet, using direct autoreconf instead of buildsystem: autotools. Is it because the buildsystem approach did not work with the github enet release link? I saw it is a bit lighter that the bespin version (does not include the docs).
> 
> The build still uses autotools; `buildsystem: autotools` is default if nothing else is declared. The autoreconf command is needed when downloading the sources from GitHub but not for the website download. To be honest, I cannot remember why I changed that to GitHub – if you prefer, I can revert it.

Ok I understand, thanks! The GitHub link is fine, it saves the users some bandwidth :D
